code/modules/projectiles/guns/energy/special.dm

```diff
@@ -20,6 +22,7 @@
 	desc = "The MK.II Prototype Ion Projector is a lightweight carbine version of the larger ion rifle, built to be ergonomic and efficient."
 	icon_state = "ioncarbine"
 	worn_icon_state = "ioncarbine"
+	worn_icon = 'icons/mob/clothing/belt.dmi'	//Carbine doesn't have a belt_mirror state so use the available state if in suit storage

```

Comment:
this is already handled by `build_worn_icon` via the slot flag, remove it.

Reply:
Reverted. Had the worn_icon set to make the other slots use a working icon. I'll add it to the list of missing textures.